### ISSUE TITLE
Testsuite for folly (master HEAD) fails

### ISSUE THREAD

**Issue by nitingupta910** [facebook/proxygen#79]
```
latest commit:
commit ed8ed6fa2bcf2effd98f2db132bd147da358ab52
Author: Alan Frindell <afrind@fb.com>
Date:   Tue Jan 26 13:19:10 2016 -0800

    More detail in HTTPCodecPrinter
    
    Summary: Print the URL or status code.
    
    Reviewed By: yfeldblum
    
    Differential Revision: D2851996
    
    fb-gh-sync-id: eed874f88164be7387de23a381bd78f8bbab1f52




=======================================================
   proxygen 32.0: lib/http/codec/test/test-suite.log
=======================================================

# TOTAL: 1
# PASS:  0
# SKIP:  0
# XFAIL: 0
# FAIL:  1
# XPASS: 0
# ERROR: 0

.. contents:: :depth: 2

FAIL: CodecTests
================

[==========] Running 55 tests from 5 test cases.
[----------] Global test environment set-up.
[----------] 2 tests from DefaultFlowControl
[ RUN      ] DefaultFlowControl.flow_control_construct

[WARNING] gmock-1.7.0/gtest/src/gtest-death-test.cc:825:: Death tests use fork(), which is unsafe particularly in a threaded context. For this test, Google Test couldn't detect the number of threads.
[       OK ] DefaultFlowControl.flow_control_construct (98 ms)
[ RUN      ] DefaultFlowControl.send_update
[       OK ] DefaultFlowControl.send_update (1 ms)
[----------] 2 tests from DefaultFlowControl (99 ms total)

[----------] 2 tests from BigWindow
[ RUN      ] BigWindow.recv_too_much
[       OK ] BigWindow.recv_too_much (2 ms)
[ RUN      ] BigWindow.remote_increase
[       OK ] BigWindow.remote_increase (0 ms)
[----------] 2 tests from BigWindow (2 ms total)

[----------] 3 tests from HTTPChecksTest
[ RUN      ] HTTPChecksTest.send_trace_body_death

[WARNING] gmock-1.7.0/gtest/src/gtest-death-test.cc:825:: Death tests use fork(), which is unsafe particularly in a threaded context. For this test, Google Test couldn't detect the number of threads.
[       OK ] HTTPChecksTest.send_trace_body_death (91 ms)
[ RUN      ] HTTPChecksTest.send_get_body
[       OK ] HTTPChecksTest.send_get_body (0 ms)
[ RUN      ] HTTPChecksTest.recv_trace_body
[       OK ] HTTPChecksTest.recv_trace_body (0 ms)
[----------] 3 tests from HTTPChecksTest (91 ms total)

[----------] 43 tests from SPDYCodecTest
[ RUN      ] SPDYCodecTest.JunkSPDY
E0127 09:17:12.988301 27564 SPDYCodec.cpp:250] SYN_STREAM: invalid length 4 < 10
[       OK ] SPDYCodecTest.JunkSPDY (0 ms)
[ RUN      ] SPDYCodecTest.LongPing
E0127 09:17:12.988493 27564 SPDYCodec.cpp:242] PING: invalid length 5 != 4
[       OK ] SPDYCodecTest.LongPing (0 ms)
[ RUN      ] SPDYCodecTest.BadType
[       OK ] SPDYCodecTest.BadType (0 ms)
[ RUN      ] SPDYCodecTest.SynStreamBoundaries
[       OK ] SPDYCodecTest.SynStreamBoundaries (27 ms)
[ RUN      ] SPDYCodecTest.SetSettings
[       OK ] SPDYCodecTest.SetSettings (0 ms)
[ RUN      ] SPDYCodecTest.FrameTooLarge
E0127 09:17:13.015197 27564 SPDYCodec.cpp:307] excessive frame size length_=1050
[       OK ] SPDYCodecTest.FrameTooLarge (0 ms)
[ RUN      ] SPDYCodecTest.FrameUncompressedTooLarge
E0127 09:17:13.015252 27564 GzipHeaderCodec.cpp:342] Decompressed headers too large
[       OK ] SPDYCodecTest.FrameUncompressedTooLarge (0 ms)
[ RUN      ] SPDYCodecTest.UnsupportedVersion
E0127 09:17:13.015290 27564 SPDYCodec.cpp:250] SYN_STREAM: invalid length 4 < 10
[       OK ] SPDYCodecTest.UnsupportedVersion (0 ms)
[ RUN      ] SPDYCodecTest.UnsupportedFrameType
[       OK ] SPDYCodecTest.UnsupportedFrameType (0 ms)
[ RUN      ] SPDYCodecTest.DefaultMaxTransactions
[       OK ] SPDYCodecTest.DefaultMaxTransactions (1 ms)
[ RUN      ] SPDYCodecTest.NonDefaultMaxTransactions
[       OK ] SPDYCodecTest.NonDefaultMaxTransactions (0 ms)
[ RUN      ] SPDYCodecTest.EmptyHeaderValue
[       OK ] SPDYCodecTest.EmptyHeaderValue (0 ms)
[ RUN      ] SPDYCodecTest.SynStreamWrongVersion
E0127 09:17:13.017009 27564 SPDYCodec.cpp:285] Invalid version=2
[       OK ] SPDYCodecTest.SynStreamWrongVersion (0 ms)
[ RUN      ] SPDYCodecTest.ShortSynReply
[       OK ] SPDYCodecTest.ShortSynReply (0 ms)
[ RUN      ] SPDYCodecTest.SupportsSessionFlowControl
[       OK ] SPDYCodecTest.SupportsSessionFlowControl (0 ms)
[ RUN      ] SPDYCodecTest.HeaderWithManyValues
SPDYCodecTest.cpp:538: Failure
Value of: 1
Expected: callbacks.headersComplete
Which is: 0
SPDYCodecTest.cpp:540: Failure
Value of: 0
Expected: callbacks.streamErrors
Which is: 1
F0127 09:17:13.017393 27564 SPDYCodecTest.cpp:542] Check failed: 'callbacks.msg.get()' Must be non NULL 
*** Check failure stack trace: ***
    @     0x7f25f7e425cd  google::LogMessage::Fail()
    @     0x7f25f7e44433  google::LogMessage::SendToLog()
    @     0x7f25f7e4215b  google::LogMessage::Flush()
    @     0x7f25f7e44e1e  google::LogMessageFatal::~LogMessageFatal()
    @           0x45534e  SPDYCodecTest_HeaderWithManyValues_Test::TestBody()
    @           0x4f8353  testing::internal::HandleExceptionsInMethodIfSupported<>()
    @           0x4f0b4a  testing::Test::Run()
    @           0x4f0c98  testing::TestInfo::Run()
    @           0x4f0d75  testing::TestCase::Run()
    @           0x4f103f  testing::internal::UnitTestImpl::RunAllTests()
    @           0x4f8833  testing::internal::HandleExceptionsInMethodIfSupported<>()
    @           0x4f1364  testing::UnitTest::Run()
    @           0x40e221  main
    @     0x7f25f72e3a00  __libc_start_main
    @           0x40e6d9  _start
    @              (nil)  (unknown)
FAIL CodecTests (exit status: 134)
```

**Comment by w-o-o** [comment]
i cannot reproduce your problem, could you paste your os, compiler and libs info? 


```
Running 55 tests
      ✓ proxygen/lib/http/codec/test:CodecTests - SPDYCodecTest.NonDefaultMaxTransactions 0.012s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - SPDYCodecTest.BadNumNameValues 0.001s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - BigWindow.recv_too_much 0.015s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - SPDYCodecTest.DefaultMaxTransactions 0.012s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - SPDYCodecTest.HeaderWithManyValues 0.007s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - SPDYCodecTest.SegmentedHeaderBlock 0.012s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - BigWindow.remote_increase 0.002s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - SPDYCodecTest.FrameTooLarge 0.000s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - DefaultFlowControl.send_update 0.001s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - SPDYCodecTest.StatusReason 0.001s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - SPDYCodecTest.LargeFrameEncoding 0.005s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - SPDYCodecTest.FrameUncompressedTooLarge 0.001s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - SPDYCodecTest.ColonHeaders 0.001s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - SPDYCodecTest.HeaderWithFin 0.000s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - SPDYCodecTest.DownstreamPing 0.000s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - SPDYCodecTest.UpstreamPing 0.000s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - SPDYCodecTest.SynStreamWrongVersion 0.000s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - SPDYCodecTest.UnsupportedFrameType 0.000s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - SPDYCodecTest.EmptyHeaderName 0.001s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - SPDYCodecTest.ServerPushHostMissing 0.000s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - SPDYCodecTest.ServerPush 0.001s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - SPDYCodecTest.ServerPushInvalidId 0.001s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - SPDYCodecTest.UnsupportedVersion 0.001s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - SPDYCodecTest.ShortSettings 0.001s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - SPDYCodecTest.ServerPushInvalidFlags 0.000s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - HTTPChecksTest.send_get_body 0.001s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - HTTPChecksTest.recv_trace_body 0.000s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - SPDYCodecTest.BadType 0.000s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - SPDYCodecTest.SendRstParsingFrame 0.001s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - SPDYCodecTest.DoubleGoawayClient 0.000s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - SPDYCodecTest.OddHeaderListTest 0.001s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - SPDYCodecTest.JunkSPDY 0.001s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - SPDYCodecTest.LongPing 0.000s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - SPDYCodecTest.DoubleGoawayServer 0.001s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - HTTP1xCodecTest.TestChunkedHeaders 0.001s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - HTTP1xCodecTest.TestSimpleHeaders 0.001s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - SPDYCodecTest.ServerPushWithoutAssoc 0.001s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - SPDYCodecTest.ShortSynReply 0.000s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - HTTP1xCodecTest.TestChunkedUpstream 0.001s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - SPDYCodecTest.SetSettings 0.000s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - SPDYCodecTest.SynStreamBoundaries 0.148s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - SPDYCodecTest.InvalidSettings 0.000s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - SPDYCodecTest.EmptyHeaderValue 0.001s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - SPDYCodecTest.HeaderDoS 0.001s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - SPDYCodecTest.DateHeader 0.000s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - SPDYCodecTest.SupportsSessionFlowControl 0.001s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - HTTP1xCodecTest.TestHeadRequestChunkedResponse 0.001s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - SPDYCodecTest.BadNVBlock 0.001s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - SPDYCodecTest.SingleGoawayClient 0.000s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - HTTP1xCodecTest.TestGetRequestChunkedResponse 0.001s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - SPDYCodecTest.StreamIdOverflow 0.000s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - SPDYCodecTest.ServerPushWithStatus 0.001s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - SPDYCodecTest.MultiplePaths 0.001s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - HTTPChecksTest.send_trace_body_death 0.261s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - DefaultFlowControl.flow_control_construct 0.266s (PASS)
      ✓ proxygen/lib/http/codec/test:CodecTests - main 0.000s (PASS)
Summary (total time 0.33s):
  PASS: 56
  FAIL: 0
  SKIP: 0
  FATAL: 0
  TIMEOUT: 0
  OMIT: 0

```

**Comment by nitingupta910** [comment]
> i cannot reproduce your problem, could you paste your os, compiler and libs info?

os: ubuntu 16.04 alpha
compiler: gcc 5.3.1
libs: folly and wangle master branch: as fetched by ./deps.sh script in proxygen

**Comment by cronnosli** [comment]
Months ago I had problems to compile proxygen by ./deps.sh
there was something very strange with what the code expected to be the google log libraries and what debian like systems installed.

I resolved it by clone and compile each one at time. So I was able to package the wangle, folly and proxygen libs and install when I needed.

I guess I have some scripts I developed to this. I'll search for them.
